sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -1254,6 +1255,1196 @@ raw_send_store(simd<T1, n1> msgSrc0, uint32_t exDesc, uint32_t msgDesc,
 }
 /// @} sycl_esimd_raw_send
 
+/// @defgroup sycl_esimd_memory_lsc LSC memory access APIs.
+/// @ingroup sycl_esimd_memory
+
+/// @addtogroup sycl_esimd_memory_lsc
+/// @{
+
+namespace detail {
+// Compute the data size for 2d block load or store.
+template <typename T, int NBlocks, int Height, int Width, bool Transposed,
+          bool Transformed>
+constexpr int get_lsc_block_2d_data_size() {
+  if (Transformed)
+    return detail::roundUpNextMultiple<Height, 4 / sizeof(T)>() *
+           detail::getNextPowerOf2<Width>() * NBlocks;
+  return Width * Height * NBlocks;
+}
+} // namespace detail
+
+/// SLM gather.
+/// Supported platforms: XeHP_SDV, DG2, PVC
+/// VISA instruction: lsc_load.slm
+///
+/// Collects elements located at slm and returns them
+/// as a single \ref simd object.
+///
+/// @tparam T is element type.
+/// @tparam NElts is the number of elements to load per address.
+/// @tparam DS is the data size.
+/// @tparam L1H is L1 cache hint.
+/// @tparam L3H is L3 cache hint.
+/// @tparam N is the number of channels (platform dependent).
+/// @param offsets is the zero-based offsets for SLM buffer in bytes.
+/// @param pred is predicates.
+/// @return is a vector of type T and size N * NElts
+///
+template <typename T, uint8_t NElts = 1,
+          lsc_data_size DS = lsc_data_size::default_size,
+          cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
```

Comment:
Cache hints don't apply for SLM accesses -- should these template parameters be removed?

Reply:
Removed cache hints from user-visible SLM APIs.